### PR TITLE
HeapSort in C

### PR DESCRIPTION
**Pull request by aaadddiii**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

File: C/sort/HeapSort.c [hunk 1]
```diff
@@ -0,0 +1,168 @@
+// C program to sort a the given numbers
+
+#include <math.h>
```
Comment (atarax665) [inline C/sort/HeapSort.c]: Remove unnecessary header files.

File: C/sort/HeapSort.c [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    printf("The sorted list :\n");
+    for (int i = 1; i < n + 1; i++)
+        printf("%d ", ar[i]);
+    return 0;
+}
+
+/*
+Working 
+Enter the number of elements to sort
+input -- >> 5
```
Comment (atarax665) [inline C/sort/HeapSort.c]:
```suggestion
 5
```

File: C/sort/HeapSort.c [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    // printing the result
+    printf("The sorted list :\n");
+    for (int i = 1; i < n + 1; i++)
+        printf("%d ", ar[i]);
+    return 0;
+}
+
+/*
+Working
```
Comment (atarax665) [inline C/sort/HeapSort.c]:
```suggestion
Sample I/O :
```

File: C/sort/HeapSort.c [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+/*
+Working 
+Enter the number of elements to sort
+input -- >> 5
+Enter the number of elements to sort
+4 3 2 5 1
+The sorted list :
+1 2 3 4 5
+*/
```
Comment (atarax665) [inline C/sort/HeapSort.c]: Add a line at the EOF.

File: C/sort/HeapSort.c [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    {
+        ar[i] = b[i];
+    }
+}
+
+int main()
+{
+    int n;
+    printf("Enter the number of elements to sort\n");
+    // input n --> number of digits in the sequence
```
Comment (atarax665) [inline C/sort/HeapSort.c]: You can remove trivial comments such as explaining I/O statements.

File: C/sort/HeapSort.c [hunk 6]
```diff
@@ -0,0 +1,168 @@
+// C program to sort a the given numbers
```
Comment (atarax665) [inline C/sort/HeapSort.c]: Add a beginner introduction of heap sort in commented form.

File: C/sort/HeapSort.c [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+/*
+Working 
+Enter the number of elements to sort
+input -- >> 5
+Enter the number of elements to sort
+4 3 2 5 1
+The sorted list :
+1 2 3 4 5
```
Comment (atarax665) [inline C/sort/HeapSort.c:168]: Do add space and time complexity of the algorithm.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: C/sort/HeapSort.c (hunk 7 above)
Comment (aaadddiii) [reply C/sort/HeapSort.c:168]: Sure thankyou

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**